### PR TITLE
Linked NDK build and various cosmetic changes

### PR DESCRIPTION
**Pull request by bitblaster**
I added the NDK build dependency to gradle to automate the c++ compilation.
I made also a bunch of cosmetic c++ changes to vsp-pty.cpp.

### CODE REVIEW

File: app/app/src/main/jni/vsp-pty.cpp [hunk 1]
```diff
@@ -139,30 +138,26 @@ extern "C"
...
 
         gEnv->CallVoidMethod(storeWeakListener, stringCallback, object);
         gEnv->DeleteLocalRef(object);
 
         if (gEnv->ExceptionCheck())
-        {
             gEnv->ExceptionDescribe();
-        }
 
-        if (getEnvStat != JNI_EDETACHED) {
```
Comment (bitblaster) [inline app/app/src/main/jni/vsp-pty.cpp:152]: This is not necessary, DetachCurrentThread guarantees no-op if there's nothing to detach (as attach does)

File: app/app/src/main/jni/vsp-pty.cpp [hunk 2]
```diff
@@ -195,79 +190,53 @@ static void* ptyThread(void* irrelevant)
         {
             //            buf[nread] = '\0';
 
-            if (CHECK_BIT(*buf, TIOCPKT_DATA))
```
Comment (bitblaster) [inline app/app/src/main/jni/vsp-pty.cpp:198]: Here *buf must be compared directly with the defined constants, they are not bit masks

File: app/app/src/main/jni/vsp-pty.cpp [hunk 3]
```diff
@@ -195,79 +190,53 @@ static void* ptyThread(void* irrelevant)
...
-            {
+            if (*buf == TIOCPKT_NOSTOP)
                 __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "TIOCPKT_NOSTOP\n");
-            }
 
-            struct termios tio;
+            struct termios tio={};
             tcgetattr(master, &tio);
             __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "Baudrate: %d\n", cfgetospeed(&tio));
-            if (tio.c_cflag & CS8)
```
Comment (bitblaster) [inline app/app/src/main/jni/vsp-pty.cpp:241]: These CS* bit masks actually must be filtered with CSIZE before comparison

File: app/local.properties [hunk 4]
```diff
@@ -1,8 +0,0 @@
-## This file must *NOT* be checked into Version Control Systems,
```
Comment (bitblaster) [inline app/local.properties:1]: I deleted this file since this should not be at all in the VCS, since it contains local machine settings

**Review by bitblaster — COMMENTED** [review]
Just some explanations on some modification I made in the pull request

File: app/app/src/main/jni/vsp-pty.cpp (hunk 3 above)
Comment (feelfreelinux) [reply app/app/src/main/jni/vsp-pty.cpp:241]: Actually most of those calls do nothing right now. I should probably do a larger refactor here. The idea was to get parity, stop bits, and bit size parameters from here and pass them through JNI, but for now I'm just passing the raw values of termios inside the `SerialData` object.

**Review by feelfreelinux — APPROVED** [review]
Thanks a lot for this contribution :)

**Comment by feelfreelinux** [comment]
CI fail unrelated.

**Pull request merged**